hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/ErrorTableUtils.java

```diff
@@ -47,27 +48,29 @@ public static Option<BaseErrorTableWriter> getErrorTableWriter(HoodieStreamer.Co
                                                                  SparkSession sparkSession,
                                                                  TypedProperties props,
                                                                  HoodieSparkEngineContext hoodieSparkContext,
-                                                                 FileSystem fileSystem) {
+                                                                 FileSystem fs,
+                                                                 Option<HoodieIngestionMetrics> metrics) {
     String errorTableWriterClass = props.getString(ERROR_TABLE_WRITE_CLASS.key());
     ValidationUtils.checkState(!StringUtils.isNullOrEmpty(errorTableWriterClass),
-        "Missing error table config " + ERROR_TABLE_WRITE_CLASS);
+                               "Missing error table config " + ERROR_TABLE_WRITE_CLASS);
 
-    Class<?>[] argClassArr = new Class[] {HoodieStreamer.Config.class,
-        SparkSession.class, TypedProperties.class, HoodieSparkEngineContext.class,
-        FileSystem.class};
-    String errMsg = "Unable to instantiate ErrorTableWriter with arguments type "
-        + Arrays.toString(argClassArr);
-    ValidationUtils.checkArgument(
-        ReflectionUtils.hasConstructor(BaseErrorTableWriter.class.getName(), argClassArr, false),
-        errMsg);
+    Class<?>[] legacyArgClass = new Class[]{HoodieStreamer.Config.class,
+        SparkSession.class, TypedProperties.class, HoodieSparkEngineContext.class, FileSystem.class};
+    Class<?>[] argClassV1 = new Class[]{HoodieStreamer.Config.class,
+        SparkSession.class, TypedProperties.class, HoodieSparkEngineContext.class, FileSystem.class, Option.class};
```

Comment:
Should we make these arg classes static as well?

Reply:
I think this is ok.